modules/http/src/main/java/org/mule/module/http/internal/request/grizzly/GrizzlyHttpClient.java

```diff
@@ -234,6 +234,12 @@ protected final ProxyServer buildProxy(ProxyConfig proxyConfig)
         {
             proxyServer = new ProxyServer(proxyConfig.getHost(),proxyConfig.getPort());
         }
+
+        if (proxyConfig.getNonProxyHosts() != null && !proxyConfig.getNonProxyHosts().isEmpty()) {
+            for (final String host : proxyConfig.getNonProxyHosts().split("\\|")) {
```

Comment:
Also, you should account for spaces in the configured String so you should trim the value.